app/models/manageiq/providers/openstack/cloud_manager/vm/resize.rb

```diff
@@ -1,9 +1,18 @@
 module ManageIQ::Providers::Openstack::CloudManager::Vm::Resize
-  def validate_resize
-    msg = validate_vm_control
-    return {:available => msg[0], :message => msg[1]} unless msg.nil?
-    return {:available => true, :message => nil} if %w(ACTIVE SHUTOFF).include?(raw_power_state)
-    {:available => false, :message => _("The Instance cannot be resized, current state has to be active or shutoff.")}
+  extend ActiveSupport::Concern
+
+  included do
+    supports :resize do
+      msg = validate_vm_control
```

Comment:
There is one issue with validate_vm_control method. In some case it returns false, in some cases true and in other cases nil.

The functions that call validate_vm_control method depend upon the "nil" value to decide whether to proceed checking next conditions or not. If we replace this by supports_control method, the next conditions in caller methods will become unreachable because supports_control will either return true/false, eliminating the flexibility "nil" provided in original validate_vm_control method.

Now the question is, what should be done in such cases where validate_vm_control returned nil? One such example of caller method can be found in "validate_shutdown_guest" method in ./app/models/manageiq/providers/vmware/infra_manager/vm/operations/guest.rb.